Update contributing guide (#1563)
iTwin/stratakit: CONTRIBUTING.md

Also, there’s [an internal package](./internal) which is used for configuration files and common variables for the workspace at large.

### Customizing Material UI

> StrataKit uses [themed components](https://mui.com/material-ui/customization/theme-components/) and [customized component structures](https://mui.com/material-ui/customization/overriding-component-structure/) to align the default Material UI components with the StrataKit design system.

Reviewer: ```suggestion
The `@stratakit/mui` package uses [themed components](https://mui.com/material-ui/customization/theme-components/) and [customized component structures](https://mui.com/material-ui/customization/overriding-component-structure/) to align the default Material UI components with the StrataKit design system.
```

Author: Updated in https://github.com/iTwin/stratakit/pull/1563/commits/b7ae630a5435fe3c713109f8e20f96c2de4ab706

Reviewer: Correct commit is [`170fa17`](https://github.com/iTwin/stratakit/pull/1563/commits/170fa17f995c6d46d21a80db3a74628501fe3fb0).

Also, there’s [an internal package](./internal) which is used for configuration files and common variables for the workspace at large.

### Customizing Material UI

StrataKit uses [themed components](https://mui.com/material-ui/customization/theme-components/) and [https://mui.com/material-ui/customization/overriding-component-structure/](customized component structures) to align the default Material UI components with the StrataKit design system.

> ### Style Guide

Reviewer: This (style guide) should probably go in a separate file?

Author: Addressed in https://github.com/iTwin/stratakit/pull/1563/commits/f0960ecfed6ceffdb1b9bb1c1194e2019a575a59

To enable us to quickly review and accept your pull requests, always create one pull request per issue. Never merge multiple requests in one unless they have the same root cause. Be sure to follow best practices and keep code changes as small as possible. Avoid pure formatting changes or random "fixes" that are unrelated to the linked issue.

### Responding to reviews

> When updating the pull request based on feedback, link to the commit that resolves the comment and allow the reviewer to choose to resolve the comment if they are satisfied.

Reviewer: Worth mentioning:
- Don't resolve threads started by others.
- Keep your branch up-to-date with the latest changes from `main`.

Author: Updated in https://github.com/iTwin/stratakit/pull/1563/commits/b7ae630a5435fe3c713109f8e20f96c2de4ab706

Reviewer: Correct commit is [`170fa17`](https://github.com/iTwin/stratakit/pull/1563/commits/170fa17f995c6d46d21a80db3a74628501fe3fb0).

### Merging

> Pull requests should be merged after at least two approvals from core team members. It is preferred for core team members to merge their own PRs. This is not a hard rule and it may be acceptable to merge with only one approval or by other core team members.

Reviewer: ```suggestion
Pull requests should be merged after at least two approvals from core team members. Core team members are expected to merge their own PRs. This is not a hard rule and it may be acceptable to merge with only one approval or by other core team members.

```

Author: Updated in https://github.com/iTwin/stratakit/pull/1563/commits/b7ae630a5435fe3c713109f8e20f96c2de4ab706

To enable us to quickly review and accept your pull requests, always create one pull request per issue. Never merge multiple requests in one unless they have the same root cause. Be sure to follow best practices and keep code changes as small as possible. Avoid pure formatting changes or random "fixes" that are unrelated to the linked issue.

### Responding to reviews

When updating the pull request based on feedback, link to the commit that resolves the comment and allow the reviewer to choose to resolve the comment if they are satisfied.

### Merging

#### Non-core contributors

Non-core team member contributors should have their pull requests reviewed and merged by a core team member.

> #### Core Team members
> 
> Team members should merge their own pull requests once it has two approvals. This is not a hard rule and it may be acceptable to merge with only one review or merge another member's pull request depending on the circumstances.
> 

Reviewer 2: This is lengthy, the main useful info here is having 2 approvals AND merging your own PR (when possible).

```suggestion
### Merging

Pull requests should be merged after at least two approvals from core team members. It is preferred for core team members to merge their own PRs. This is not a hard rule and it may be acceptable to merge with only one approval or by other core team members.
```
